administration/scheduling-and-retries.md

```diff
@@ -10,6 +10,41 @@ Once an output plugin gets called to flush some data, after processing that data
 
 If the return status was **OK**, it means it was successfully able to process and flush the data, if it returned an **Error** status, means that an unrecoverable error happened and the engine should not try to flush that data again. If a **Retry** was requested, the _Engine_ will ask the _Scheduler_ to retry to flush that data, the Scheduler will decide how many seconds to wait before that happens.
 
+## Configuring Wait Time for Retry
+
+The Scheduler provides two configuration options called **scheduler.cap** and **scheduler.base** which can be set on service section. 
+
+| Key | Description | Default Value | 
+| -- | ------------| --------------| 
+| scheduler.cap | Set a maximum retry time in second. The property is supported from v1.8.7. | 2000 | 
+| scheduler.base | Set a base of exponential backoff. The property is supported from v1.8.7. | 5 |
+
+These two configuration determine the waiting time before retry happen. Fluent Bit use exponential-backoff-and-jitter algorithm to determine the waiting time. These configuration means when the Nth retry needed, scheduler will generate a random number X between `(min(base * (Nth power of 2), cap), cap)` as waiting time for this retry.Basically, the **scheduler.base** decide how fast retry could happen, the **scheduler.cap** decide what's the longest waiting time retry could be. 
+
+For detail explanation on exponential-backoff-and-jitter algorithm, please check this [blog](https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/).
+
+### Example
+
+The following example configure the **scheduler.base** as 5 seconds and **scheduler.cap** as 16 seconds. 
+
+```text
+[SERVICE]
+    Flush            5
+    Daemon           off
+    Log_Level        debug
+    scheduler.base   5
+    scheduler.cap    16
+```
+
+The waiting time will be:
+
+| Nth retry | waiting time range (seconds) |
+| --- | --- | 
+| 1 | (5, 16)   |
+| 2 | (5*2, 16) |
+| 3 | (16, 16)  |
+| 4 | (16, 16)  |
```

Comment:
Hmm, are you sure about this? From the blog post you submitted it seems like jitter at least in AWS' context means we will always select a random number between 0 and the current exponentually backed off wait time. So it would be 

0-5
0-10
0-16
0-16

Now I'm not sure if that is what the fluent bit code is doing but I believe that is what the amazon article expects it to do. Please let me know if you have a different understanding.

Reply:
Yeah you are right, it should be the upper bound keep changing until reach to the maximum, the lower bound will be the base. Which the range should be:
5 - 10
5 - 16
5 - 16
5 - 16

Reply:
The logic looks good now.